pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java

```diff
@@ -88,25 +89,51 @@ public Representation post(Representation entity) {
   @Tags({ "table" })
   @Paths({ "/tables", "/tables/" })
   private void addTable(AbstractTableConfig config) throws IOException {
+    ensureMinReplicas(config);
+
+    _pinotHelixResourceManager.addTable(config);
+  }
 
+  private void ensureMinReplicas(AbstractTableConfig config) {
     // For self-serviced cluster, ensure that the tables are created with atleast
     // min replication factor irrespective of table configuratio value.
     SegmentsValidationAndRetentionConfig segmentsConfig = config.getValidationConfig();
-    int requestReplication;
-    try {
-      requestReplication = segmentsConfig.getReplicationNumber();
-    } catch (NumberFormatException e) {
-      throw new PinotHelixResourceManager.InvalidTableConfigException("Invalid replication number", e);
-    }
     int configMinReplication = _controllerConf.getDefaultTableMinReplicas();
-    if (requestReplication < configMinReplication) {
-      LOGGER.info("Creating table with minimum replication factor of: {} instead of requested replication: {}",
-          configMinReplication, requestReplication);
-      segmentsConfig.setReplication(String.valueOf(configMinReplication));
+    boolean verifyReplicasPerPartition = false;
+    boolean verifyReplication = true;
+
+    if (TableType.REALTIME.name().equalsIgnoreCase(config.getTableType())) {
+      KafkaStreamMetadata kafkaStreamMetadata;
+      try {
+        kafkaStreamMetadata = new KafkaStreamMetadata(config.getIndexingConfig().getStreamConfigs());
+      } catch (Exception e) {
+        throw new PinotHelixResourceManager.InvalidTableConfigException("Invalid tableIndexConfig or streamConfigs", e);
+      }
+      verifyReplicasPerPartition = kafkaStreamMetadata.hasSimpleKafkaConsumerType();
+      verifyReplication = kafkaStreamMetadata.hasHighLevelKafkaConsumerType();
     }
 
-    String replicasPerPartitionStr = segmentsConfig.getReplicasPerPartition();
-    if (replicasPerPartitionStr != null) {
+    if (verifyReplication) {
+      int requestReplication;
+      try {
+        requestReplication = segmentsConfig.getReplicationNumber();
+      } catch (NumberFormatException e) {
+        throw new PinotHelixResourceManager.InvalidTableConfigException("Invalid replication number", e);
+      }
+      if (requestReplication < configMinReplication) {
+        LOGGER.info("Creating table with minimum replication factor of: {} instead of requested replication: {}",
+            configMinReplication, requestReplication);
+        segmentsConfig.setReplication(String.valueOf(configMinReplication));
+      }
+    } else {
+      throw new PinotHelixResourceManager.InvalidTableConfigException("Invalid table type: '" + config.getTableType() + "'");
```

Comment:
Maybe I'm missing something, but doesn't that mean that this will always throw an exception if the table type is LLC-only (`verifyReplication=false`)